src/NServiceBus.Persistence.AzureStorage/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -101,20 +101,41 @@ static string DecodeFrom64(string encodedData)
         /// <returns>Subscription addresses that were found for the provided messageTypes</returns>
         public async Task<IEnumerable<Subscriber>> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes, ContextBag context)
         {
-            var subscribers = new List<Subscriber>();
+            var subscribers = new HashSet<Subscriber>(new SubscriberComparer());
             var table = client.GetTableReference(subscriptionTableName);
 
             foreach (var messageType in messageTypes)
```

Comment:
Forgive my ignorance but technically we have here a N+1 query problem (although we probably never really pass multiple message types into it. Ignoring that could you write one query which gets them all?